pdc_client/runner.py

```diff
@@ -110,14 +110,19 @@ def load_plugins(self):
             for name in os.listdir(dir):
                 if not name.endswith('.py') or name not in plugins_set:
                     continue
-                file, pathname, description = imp.find_module(name[:-3], [dir])
-                plugin = imp.load_module(name[:-3], file, pathname, description)
-                self.logger.debug('Loaded plugin {0}'.format(name[:-3]))
-                self.raw_plugins.append(plugin)
-                if hasattr(plugin, 'PLUGIN_CLASSES'):
-                    for p in plugin.PLUGIN_CLASSES:
-                        self.logger.debug('Instantiating {0}'.format(p.__name__))
-                        self.plugins.append(p(self))
+                try:
+                    file, pathname, description = imp.find_module(name[:-3], [dir])
+                    plugin = imp.load_module(name[:-3], file, pathname, description)
+                    self.logger.debug('Loaded plugin {0}'.format(name[:-3]))
+                    self.raw_plugins.append(plugin)
+                    if hasattr(plugin, 'PLUGIN_CLASSES'):
+                        for p in plugin.PLUGIN_CLASSES:
+                            self.logger.debug('Instantiating {0}'.format(p.__name__))
+                            self.plugins.append(p(self))
+                except:
+                    print('Failed to load plugin {0}'.format(name[:-3]))
+                finally:
+                    file.close()
```

Comment:
Use `if file:` here or `with file:` after getting the file.

Reply:
changed, but do you think it's possible to get a None value of file here?